src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -79,6 +79,12 @@ def trace(self, append_to_templated: str = "") -> JinjaTrace:
         trace_entries: List[regex.Match] = list(
             regex.finditer(r"\0", trace_template_output)
         )
+        # If the file has no templated entries, we should just iterate
+        # through the raw slices to add all the placeholders.
+        if not trace_entries:
```

Comment:
This potentially feels a little odd, in that it's adding a special case. What happens in the case where there **are** trace entries, but there are raw (literal) slices preceding them? Or following them?

I worry that we could be fixing one issue while possibly missing more subtle related/remaining issues.